Modules/IO/GDCM/src/itkGDCMImageIO.cxx

```diff
@@ -1298,7 +1298,7 @@ GDCMImageIO::Write(const void * buffer)
 
     image.SetIntercept(m_RescaleIntercept);
     image.SetSlope(m_RescaleSlope);
-    auto *       copyBuffer = new char[len];
+    auto *       copyBuffer = new char[numberOfBytes];
```

Comment:
@seanm 

Have you been able to confirm:
```
pixeldata.SetByteValue(copyBuffer, static_cast<uint32_t>(numberOfBytes));
```

Reply:
I haven't had time to look again, but will this week.

Reply:
In fact, looking at it more today, I think this might be my bug, and not ITK at all... will dig deeper...

Reply:
OK, this was our bug, so I'll just close this.